### DISCUSSION TITLE
Partner Sharing starting at specific date

### DISCUSSION THREAD

**Discussion by jhbruhn** [immich-app/immich#2654] [Feature Request]
Similar to Google photos, I would like it if Partner Sharing could be enabled for photos starting at a specific date.

Additionally, maybe multiple views for only mine, my partner's and all photos could added (if not implemented already?)

**Comment by bozzfozz** [comment]
i think either you want to share all the photos or just certain ones in the albums. and this google function bothered me for example.

but everyone thinks differently

**Reply by jhbruhn** [reply]
I see your point, although at least for me it would be an easy filter for photos I maybe don't want to share (as in pre relationship for example), while still benefitting from the ease of partner sharing for new photos.

I think as long as it is optional, everyone will be happy.

**Comment by nicohansi** [comment]
I would also like to see this.  
For me the greatest benefit of immich is, that it makes albums obsolete (for me).  
My partner and I put all our photos in immich and Partner Sharing is amazing because every photo of any new event is directly there. But unfortunately the pre-relationship timeline is a complete mess of mixed events now.

**Comment by dany9328** [comment]
This would be a great feature. We've migrated from Google Photos and liked this sharing start-date.

**Comment by wientanz** [comment]
I second this, that feature would be useful, I had multiple partners over the years and while my current partner does not have an issue with this it's just not necessary for her to see 10 years of photos that have nothing to do with our family

**Comment by amvanmaren** [comment]
I would love to see this implemented.  My partner does not want to see all of my pre-relationship photos.  Currently my only solution to this is to create a separate account with the old photos and not share from that account.

**Comment by dalekirkwood** [comment]
Here to also say, we would love to see this implemented.

**Comment by ArbiterGR** [comment]
Just wanted to say that i would love to see this implemented too! My SO would certainly not want to see me in my previous relationships :P

**Comment by ildoc** [comment]
would also be useful just the possibility to create an album with a rule (eg time based like "all photos from this date" or "all photos with this person/these people")
o even the possibility of combining multiple rules

**Comment by wolterkam** [comment]
Were there any attempts made at implementing this? It's been a while and it feels like such an obvious feature that shouldn't even be too hard to get it working.

**Comment by yann117** [comment]
Partner sharing is just one specific "sharing use-case" that we hope to be refactored more generally with the huge topic of "Better sharing in Immich":
https://github.com/immich-app/immich/issues/12614

This is planned for 3.1 or 3.2, just be a little bit more patient.

"partner" can then just be seen as one relationship between two entities, among all the other ones: friends, family, colleagues, sport clubs, association, ...

**Reply by bjwo** [reply]
@yann117 But would this use-case be covered - to start sharing the library only from a specific start date?
